content/python/concepts/built-in-functions/terms/enumerate/enumerate.md

```diff
@@ -0,0 +1,102 @@
+---
+Title: 'enumerate()'
+Description: 'Returns a list of tuples containing the index and the element of all the elements in an iterator.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Iterators'
+  - 'Lists'
+  - 'Tuples'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/data-science'
+---
+
+The **`enumerate()`** function returns a list of [tuples](https://www.codecademy.com/resources/docs/python/tuples).
+
+## Syntax
+
+```pseudo
+enumerate(iterable)
+```
+
+The `iterable` parameter can be one of the following:
+
+- A [`string`](https://www.codecademy.com/resources/docs/python/strings)
+- A [`list`](https://www.codecademy.com/resources/docs/python/lists),[`tuple`](https://www.codecademy.com/resources/docs/python/tuples),[`dictionary`](https://www.codecademy.com/resources/docs/python/dictionaries), or [`set`](https://www.codecademy.com/resources/docs/python/sets).
+- An iterable object such as one returned by the [`iter()`](https://www.codecademy.com/resources/docs/python/iterators/iter) function.
+
+
+## Example
+
+The example below demonstrates how the `enumerate()` function is used on a list:
+
+```py
+companies_list = ["Google","Microsoft","Amazon"]
+for index,company in enumerate(companies_list):
+  print(f"{index}. {company}")
+```
+Enumerate returns a Tuple as shown below:
+
+```shell
+[(0,"Google"),(1,"Microsoft"),(2,"Amazon")]
+```
+
+This will print the following output:
+
+```shell
+0. Google
+1. Microsoft
+2. Amazon
+```
+
+### Codebytes
+
+The following examples show how the `enumerate()` function is used on some of the iterables:
+
+```codebyte/python
+1. #Using Enumerate on a String
+
+word = "Python"
+
+for index, letter in enumerate(word):
+    print(index, letter)
+    
+#Output:

```

Comment:
We generally do not include output for codebytes. (The user can run them to see the output.